src/utils/SecurityUtils.js

```diff
@@ -0,0 +1,206 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+const _findById = (array, idToFind) => {
+  return array.find((element) => element.id === idToFind);
+};
+
+const areAccessControlListsIdentical = (aclA, aclB) => {
+  if (Object.keys(aclA).length !== Object.keys(aclB).length) return false;
+  for (const userAccessA of aclA) {
+    const userAccessB = _findById(aclB, userAccessA.id);
+    if (userAccessB === undefined || userAccessA.role !== userAccessB.role) return false;
+  }
+  return true;
+};
+
+const compareAccessControlLists = (currentACL, newACL) => {
+  const entriesToAdd = [];
+  const entriesToRemove = [];
+  for (const newUserAccess of newACL) {
+    const currentUserAccess = _findById(currentACL, newUserAccess.id);
+    // New entry in ACL
+    if (currentUserAccess === undefined) entriesToAdd.push(newUserAccess);
+    // User role changed
+    else if (currentUserAccess.role !== newUserAccess.role) {
+      entriesToRemove.push(currentUserAccess);
+      entriesToAdd.push(newUserAccess);
+    }
+  }
+
+  for (const currentUserAccess of currentACL) {
+    const newUserAccess = _findById(newACL, currentUserAccess.id);
+    // Entry removed from ACL
+    if (newUserAccess === undefined) entriesToRemove.push(currentUserAccess);
+  }
+
+  const usersToRemove = entriesToRemove.map((entry) => entry.id);
```

Comment:
Can you reuse the function getUsersFromAcl here?

Reply:
Good point :smile: